### DISCUSSION TITLE
Quick Question about Notifications

### DISCUSSION THREAD

**Discussion by tentator** [hluk/CopyQ#2498] [General]
Is there a way to display a notification at the mouse pointer location?
Something like a tooltip..

I tried with the `notification()` function but it seems to have no '.x' or '.y' or similar coordinate attributes I could use to position it..

The only function that does something almost similar is `menuItems()` but that one has interactive item selection requirements..

Idea is to just notify something to the user, but at his mouse pointer location.. screens are getting too big nowadays.. :)

Any suggestions are welcome, thanks!

Kr,
 tent:wq

**Comment by hluk** [comment]
> I tried with the `notification()` function but it seems to have no '.x' or '.y' or similar coordinate attributes I could use to position it..

AFAIK, notifications in many desktop environment do not support custom coordinates.

> The only function that does something almost similar is `menuItems()` but that one has interactive item selection requirements..

There is also `dialog()` which allows to show a popup with various custom GUI elements. But both `menuItems()` and `dialog()` would "steal" keyboard focus unlike notifications.

Example script:

```js
const text = 'Some message!';
const pos = pointerPosition();
dialog('.x', pos[0], '.y', pos[1], '.label', text);
```

**Reply by tentator** [reply]
Thanks for the ideas, I tried it, but as you suspected there is a problem with "focus stealing".. would it be possible to implement something to avoid it (or is this OS dependant)?
Either dialog() with option not to steal focus, or the notification() with addition of .x and .y (and if the desktop environ would not honour it, well it would be just like it is now, in some corner of the screen..)
What do you think?
